File: casa_guido/app/src/main/java/br/com/casa_guido/room/entidades/SituacaoHabitacional.kt
package br.com.casa_guido.room.entidades

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.math.BigDecimal
import java.util.UUID

@Entity(tableName = "situacao_habitacional")
data class SituacaoHabitacional(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val pacienteId: String = "", //usado
    val comoAdquiriuCasa: Int? = null, //usado
    val area: Int? = null,
    val meioDeTransporte: Int? = null,
    val meioDeComunicao: Int? = null,
    val possuiBanheiros: Int? = null,
    val numeroComodos: String = "",
    val material: Int? = null,
    val bens: Array<Int> = emptyArray(),
    val dentroDeCasa: Int? = null
)

File: backend/models/SituacaoHabitacional.js
const mongoose = require('mongoose');

const SituacaoHabitacionalSchema = new mongoose.Schema({
  paciente: { type: mongoose.Schema.Types.ObjectId, ref: 'Paciente' },
  comoAdquiriuCasa: Number,
  area: Number,
  numeroComodos: Number,
  material: Number,
  bens: [Number],
  meioDeTransporte: Number,
  meioDeComunicao: Number,
  possuiBanheiros: Number
});

module.exports = mongoose.model('SituacaoHabitacional', SituacaoHabitacionalSchema); 